src/main/java/de/qabel/desktop/remoteFS/RemoteFSController.java

```diff
@@ -0,0 +1,179 @@
+package de.qabel.desktop.remoteFS;
+
+import com.amazonaws.auth.DefaultAWSCredentialsProviderChain;
+import de.qabel.core.crypto.CryptoUtils;
+import de.qabel.core.crypto.QblECKeyPair;
+import de.qabel.desktop.CellValueFactory.BoxObjectCellValueFactory;
+import de.qabel.desktop.exceptions.QblStorageException;
+import de.qabel.desktop.storage.*;
+import javafx.fxml.FXML;
+import javafx.fxml.Initializable;
+import javafx.scene.control.TreeItem;
+import javafx.scene.control.TreeTableColumn;
+import javafx.scene.control.TreeTableView;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.ResourceBundle;
+import java.util.UUID;
+
+
+public class RemoteFSController implements Initializable {
+
+    private ImageView folderIcon = new ImageView(
+            new Image(getClass().getResourceAsStream("/folder.png"))
+    );
+
+    private TreeItem<BoxObject> parentNode;
+    private List<BoxFolder> folders;
+
+    @FXML
+    private TreeTableView treeTable;
+    @FXML
+    private TreeTableColumn<BoxObject, String> nameColumn;
+    @FXML
+    private TreeTableColumn<BoxObject, String> sizeColumn;
+    @FXML
+    private TreeTableColumn<BoxObject, String> dateColumn;
+
+    @FXML
+    public void initialize(URL location, ResourceBundle resources) {
+
+
+        try {
+            BoxNavigation nav = createSetup();
+            uploadFilesAndFolders(nav);
+
+            TreeItem rootNodeWithChilds = calculateFolderStructure(nav);
+            treeTable.setRoot(rootNodeWithChilds);
+        } catch (QblStorageException e) {
+            e.printStackTrace();
+        }
+        calculateTableContent();
+
+
+    }
+
+    TreeItem calculateFolderStructure(BoxNavigation nav) throws QblStorageException {
+
+      //  BoxFolder rootFolder = new BoxFolder("block", "root Folder", new byte[16]);
+
+        TreeItem<BoxObject> rootNode = new TreeItem<>(new BoxFolder("block", "root Folder", new byte[16]));
+        parentNode = calculateSubFolderStructure(nav, rootNode, true);
+        parentNode.setExpanded(true);
+        return parentNode;
+    }
+
+    private TreeItem<BoxObject> calculateSubFolderStructure(
```

Comment:
Still, this won't scale.
Imagine a user syncing his home folder ... dozens of config folders containing dozens of other config folders with dozens of config files .... which the user needs but won't ever open in tree view. Nevertheless, all these folders get eagerly loaded by this implementation. Don't think this is acceptable from a users point of view. As JavaFX focusses on async operations, I'm pretty sure there is an easy way to perform async resolutions of TreeItems